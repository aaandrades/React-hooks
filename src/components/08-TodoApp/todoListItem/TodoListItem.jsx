import React from "react";

export const TodoListItem = ({ state, index, handleDelete, handleToggle }) => {
  return (
    <div
      key={state.id}
      className="containerOption"
      style={state.done ? { background: "#ACF7C1" } : { background: "" }}
      onClick={() => handleToggle(state.id)}
    >
      <p className="containerOption__text">
        {index + 1}. {state.desc}
      </p>
      <button className="btn btn-danger" onClick={() => handleDelete(state.id)}>
        Delete
      </button>
    </div>
  );
};
