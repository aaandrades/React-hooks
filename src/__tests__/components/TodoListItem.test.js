import { shallow } from "enzyme";
import { TodoListItem } from "../../components/08-TodoApp/todoListItem/TodoListItem";
import { demoTodos } from "../../utils/demoTodos";

jest.mock("");

describe("Test for TodoListItem", () => {
  const handleDelete = jest.fn();
  const handleToggle = jest.fn();

  const wrapper = shallow(
    <TodoListItem
      state={demoTodos[0]}
      index={0}
      handleDelete={handleDelete}
      handleToggle={handleToggle}
    />
  );

  it("should create correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call delete function", () => {
    wrapper.find("button").simulate("click");
    expect(handleDelete).toHaveBeenCalledWith(demoTodos[0].id);
  });

  it("should call toggle function", () => {
    wrapper.find(".containerOption").simulate("click");
    expect(handleToggle).toHaveBeenCalled();
  });

  it("should show the text correctly", () => {
    const p = wrapper.find("p");
    expect(p.text()).toBe(1 + ". " + demoTodos[0].desc);
  });

});
