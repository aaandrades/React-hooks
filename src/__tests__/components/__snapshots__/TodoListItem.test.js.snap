// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for TodoListItem should create correctly 1`] = `
<div
  className="containerOption"
  key="1"
  onClick={[Function]}
  style={
    Object {
      "background": "",
    }
  }
>
  <p
    className="containerOption__text"
  >
    1
    . 
    Learn React
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
